servatrice/src/serversocketinterface.cpp

```diff
@@ -1000,3 +1002,54 @@ Response::ResponseCode ServerSocketInterface::cmdReloadConfig(const Command_Relo
     settingsCache->sync();
     return Response::RespOk;
 }
+
+Response::ResponseCode ServerSocketInterface::cmdAdjustMod(const Command_AdjustMod &cmd, ResponseContainer & /*rc*/) {
+
+    QString userName = QString::fromStdString(cmd.user_name());
+
+    if (cmd.should_be_mod()) {
+        logDebugMessage("Received admin command: promote user");
+        QSqlQuery *query = sqlInterface->prepareQuery(
+                "update {prefix}_users set admin = :adminlevel where name = :username");
+        query->bindValue(":adminlevel", 2);
+        query->bindValue(":username", userName);
+        if (!sqlInterface->execSqlQuery(query))
+            return Response::RespInternalError;
+
+        ServerSocketInterface *user = static_cast<ServerSocketInterface *>(server->getUsers().value(userName));
+        Event_NotifyUser event;
+        event.set_type(Event_NotifyUser::PROMOTED);
+        SessionEvent *se = user->prepareSessionEvent(event);
+        user->sendProtocolItem(*se);
+        delete se;
+
+    } else {
+        logDebugMessage("Received admin command: demote user");
+        QSqlQuery *query = sqlInterface->prepareQuery("update {prefix}_users set admin = :adminlevel where name = :username");
+        query->bindValue(":adminlevel", 0);
+        query->bindValue(":username", userName);
+        if (!sqlInterface->execSqlQuery(query))
+            return Response::RespInternalError;
+
+        QList<ServerSocketInterface *> userList;
+        ServerSocketInterface *user = static_cast<ServerSocketInterface *>(server->getUsers().value(userName));
+        userList.append(user);
+
+        if (!userList.isEmpty()) {
```

Comment:
An ip or id ban can apply to multiple online users, so it makes sense to have a list. This can only ever affect one user at a time though, so it doesn't make sense.